[Debug] assert compile error

### bug描述 Describe the Bug

版本 develop 76b328bcb687dc91886aba878711cd9bbc2c60ed
GCC 9.4.0

assert内存在类型错误

位置[paddle/ir/core/value.cc](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/ir/core/value.cc)
268行
```
OpResultImpl::~OpResultImpl() {
  assert(use_empty() &&
         owner()->name() + " operation destroyed but still has uses.");
}
```
原因  use_empty() 是布尔值。不可以和后面的字符串比较。

错误LOG

```
In file included from /usr/include/c++/9/cassert:44,
                 from /media/wjl/D2/github/fork/Paddle/paddle/utils/string/tinyformat/tinyformat.h:132,
                 from /media/wjl/D2/github/fork/Paddle/paddle/utils/string/printf.h:77,
                 from /media/wjl/D2/github/fork/Paddle/paddle/ir/core/enforce.h:20,
                 from /media/wjl/D2/github/fork/Paddle/paddle/ir/core/value.cc:16:
/media/wjl/D2/github/fork/Paddle/paddle/ir/core/value.cc: In destructor ‘ir::detail::OpResultImpl::~OpResultImpl()’:
/media/wjl/D2/github/fork/Paddle/paddle/ir/core/value.cc:268:22: error: no match for ‘operator&&’ (operand types are ‘bool’ and ‘std::__cxx11::basic_string<char>’)
  268 |   assert(use_empty() &&
      |          ~~~~~~~~~~~ ^~
      |                   |
      |                   bool
  269 |          owner()->name() + " operation destroyed but still has uses.");
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                          |
      |                          std::__cxx11::basic_string<char>
/media/wjl/D2/github/fork/Paddle/paddle/ir/core/value.cc:268:22: note: candidate: ‘operator&&(bool, bool)’ <built-in>
  268 |   assert(use_empty() &&
      |          ~~~~~~~~~~~~^~
  269 |          owner()->name() + " operation destroyed but still has uses.");

```

### 其他补充信息 Additional Supplementary Information

_No response_

--- a/paddle/ir/core/value.cc
+++ b/paddle/ir/core/value.cc
@@ -264,10 +264,7 @@ uint32_t OpResultImpl::GetResultIndex() const {
   return ir::dyn_cast<OpInlineResultImpl>(this)->GetResultIndex();
 }
 
-OpResultImpl::~OpResultImpl() {
-  assert(use_empty() &&
-         owner()->name() + " operation destroyed but still has uses.");
-}
+OpResultImpl::~OpResultImpl() { assert(use_empty()); }
 
 ir::Operation *OpResultImpl::owner() const {
   // For inline result, pointer offset index to obtain the address of op.

